iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/analyze/schema/AutoCreateSchemaExecutor.java

```diff
@@ -54,21 +59,33 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
-import java.util.function.Function;
 import java.util.function.IntFunction;
 import java.util.stream.Collectors;
 
 import static org.apache.iotdb.db.utils.EncodingInferenceUtils.getDefaultEncoding;
 
 class AutoCreateSchemaExecutor {
-
+  private final IoTDBConfig config = IoTDBDescriptor.getInstance().getConfig();
+  private final Coordinator coordinator;
   private final ITemplateManager templateManager;
-  private final Function<Statement, ExecutionResult> statementExecutor;
+  private final ISchemaFetcher schemaFetcher;
 
   AutoCreateSchemaExecutor(
-      ITemplateManager templateManager, Function<Statement, ExecutionResult> statementExecutor) {
+      Coordinator coordinator, ITemplateManager templateManager, ISchemaFetcher schemaFetcher) {
+    this.coordinator = coordinator;
     this.templateManager = templateManager;
-    this.statementExecutor = statementExecutor;
+    this.schemaFetcher = schemaFetcher;
+  }
+
+  private ExecutionResult executeStatement(Statement statement) {
+    return coordinator.execute(
+        statement,
+        SessionManager.getInstance().requestQueryId(),
+        null,
+        "",
+        ClusterPartitionFetcher.getInstance(),
+        schemaFetcher,
+        config.getQueryTimeoutThreshold());
```

Comment:
get timeout from context.

Reply:
This method will only be called when writing (for automatic schema creation).The type of context is WRITE, and its timeout is set to LONG.MAX_VALUE, which doesn't make any sense.